hpx/parallel/algorithms/merge.hpp

```diff
@@ -491,6 +490,341 @@ namespace hpx { namespace parallel { inline namespace v1
                 std::forward<Proj1>(proj1),
                 std::forward<Proj2>(proj2)));
     }
+
+    /////////////////////////////////////////////////////////////////////////////
+    // inplace_merge
+    namespace detail
+    {
+        /// \cond NOINTERNAL
+
+        // sequential inplace_merge with projection function.
+        template <typename BidirIter, typename Comp, typename Proj>
+        inline BidirIter
+        sequential_inplace_merge(
+            BidirIter first, BidirIter middle, BidirIter last,
+            Comp && comp, Proj && proj)
+        {
+            std::inplace_merge(first, middle, last,
+                util::compare_projected<Comp, Proj>(
+                    std::forward<Comp>(comp),
+                    std::forward<Proj>(proj)));
+            return last;
+        }
+
+        template <typename ExPolicy, typename RandIter,
+            typename Comp, typename Proj,
+        HPX_CONCEPT_REQUIRES_(
```

Comment:
Do we have to use concepts for this internal API? What is your rationale of using it here?

Reply:
@hkaiser I just wanted to restrict that ```RandIter``` is only random access iterator.
Orginally, inplace_merge have to support bidirectional iterator, too.
But, in now, it only supports random access iterator.
I thought that I might have to implement different helper function for the bidirectional iterator in the future.
In that context, I used concepts for this helper function.

If it is strange and misuse of concepts, I will modify this. Is this misuse of concepts?